### ISSUE TITLE
no concrete type registered for type URL /juno.v1.clock.UpdateParamProposal

### ISSUE THREAD

**Issue by ducphamle2** [oraichain/wasmd#4]
When using the `juno` command: `oraid tx clock add-contract`, we received the following error:

![image](https://github.com/user-attachments/assets/906f5993-9780-4032-90fc-63fa18e35edf)

After debugging for a day, the closest I have got to understand this bug is that:
- Cosmos SDK 0.50x has changed the way it handles gov proposals.
- Instead of whitelisting the gov proposal content (like `/juno.v1.clock.UpdateParamProposal`), it recursively validates the entire `MsgSubmitProposal` message, consider any sub-messages within it also to be a Cosmos message as well.
- Because `/juno.v1.clock.UpdateParamProposal` is not a message -> hence the error

**Comment by ducphamle2** [comment]
My attempt to fix the bug by registering `/juno.v1.clock.UpdateParamProposal` didn't work, even though it bypassed the above error:

```
registry.RegisterInterface("juno.clock.v1.UpdateParamsProposal", (*sdk.Msg)(nil), &UpdateParamsProposal{})
```

![image](https://github.com/user-attachments/assets/51ae354d-857e-4df5-9831-db18be33916f)


This is because there are additional checks for the message such as no `cosmos.msg.v1.signer` option found for the message.

![image](https://github.com/user-attachments/assets/f2f72559-14ef-4f9c-a5d0-bee38f07e5a8)

This is odd, because it should not be intended to be a message.

**Comment by ducphamle2** [comment]
I then switched to a new direction, by using the new official way of submitting gov proposals: https://docs.cosmos.network/main/build/modules/gov#messages

![image](https://github.com/user-attachments/assets/9f1ffd94-2bab-4ae6-ba51-4abd299912fb)

I added a new `msgServer` message, called `MsgUpdateParams` and registered it as a message. I also added its basic Msg implementations.

![image](https://github.com/user-attachments/assets/09af7902-6a86-4288-89e5-20f71dabe166)
![image](https://github.com/user-attachments/assets/628d45f0-ed52-4dc4-8f47-bd9cc91dc0a0)
<img width="700" alt="image" src="https://github.com/user-attachments/assets/4fb94bfa-33b9-4d99-b1a8-7a82e33b115b">


Fortunately, it worked.

**Comment by ducphamle2** [comment]
Please refer to these commits to see how I fixed it: https://github.com/oraichain/juno/commit/3c83f1d805c649906310c0b46878604be74db16c, https://github.com/oraichain/juno/commit/2c59cc9423748a7a196f3030997de0b8d5bda735, 

Also, fixed in #2
